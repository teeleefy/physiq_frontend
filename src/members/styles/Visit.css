.Visit-card {
    height: 30vh;
    margin-left: 0.5rem;
    margin-bottom: 0.5rem;
}

.Visit-label {
    font-size: 1.5rem;
}

.Visit-h1 {
    font-size: 3rem;
    font-weight: 600;
}

.Visits-navlink {
    text-decoration: none;
    color: white;
}