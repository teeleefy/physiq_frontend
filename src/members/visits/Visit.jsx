import {
    Card,
    CardBody,
    CardText,
    ListGroupItem
  } from "reactstrap";
  import { NavLink } from "react-router-dom";
import '../styles/Visit.css'


function Visit({visit, memberId}){

    return(
        <>
            <ListGroupItem className="Visit-card">
            <Card className="Visit-card">
                <CardBody>
                    <CardText>
                        
                        <p><b>Visit Name:</b> <NavLink className="Visit-name" to={`/member/${memberId}/visits/${visit.id}`} key={visit.id}>
                            {visit.title}
                        </NavLink></p>
                        <p><b>Date:</b> {visit.date}</p>
                        <p><b>Clinic Visited:</b> {visit.clinic}</p>
                        <p><b>Doctor Seen:</b> {visit.doctor}</p>
                        <p><b>Description:</b> {visit.description}</p>
                        <p><b>Notes:</b> {visit.notes}</p>
                    </CardText>
                </CardBody>
            </Card>
            </ListGroupItem>
        </>
     )
}

export default Visit;