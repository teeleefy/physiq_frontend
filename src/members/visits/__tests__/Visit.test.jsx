import { renderWithRouter, screen } from '../../../../__tests__/test-tools';
import Visit from '../Visit'

let visit={title: "Follow-up visit", date: "2023-04-10", doctor: "Dr. Trent", clinic: "Main Street Cardio", notes: "Followed up after open heart surgery."}
let memberId=1000;

describe('Visit', () => {
  it('renders the Visit component', () => {
    renderWithRouter(<Visit visit={visit} memberId={memberId}/>)
    
    screen.debug(); // prints out the jsx in the Visit component unto the command line
  })
})

it("matches snapshot", function() {
  const { asFragment } = renderWithRouter(<Visit visit={visit} memberId={memberId}/>);
  expect(asFragment()).toMatchSnapshot();
});