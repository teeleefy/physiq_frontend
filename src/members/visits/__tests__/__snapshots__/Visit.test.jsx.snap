// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="list-group-item"
  >
    <div
      class="Visit-card card"
    >
      <div
        class="card-body"
      >
        <p
          class="card-text"
        />
        <p>
          <b>
            Visit Name:
          </b>
           
          <a
            class="Visit-name"
            href="/member/1000/visits/undefined"
          >
            Follow-up visit
          </a>
        </p>
        <p>
          <b>
            Date:
          </b>
           2023-04-10
        </p>
        <p>
          <b>
            Doctor Seen:
          </b>
           Dr. Trent
        </p>
        <p>
          <b>
            Clinic Visited:
          </b>
           Main Street Cardio
        </p>
        <p />
      </div>
    </div>
  </li>
</DocumentFragment>
`;
